### ISSUE TITLE
Tests fail when configured with meson

### ISSUE THREAD

**Issue by vedranmiletic** [gabime/spdlog#1295]
This is on FreeBSD but I have observed similar behavior on Linux, with both v1.4.2 and master:

```
% meson -Denable_benchmarks=true -Denable_examples=true -Denable_tests=true builddir-meson                           
The Meson build system                                                                                             
Version: 0.51.2                                                                                                    
Source dir: /usr/home/vedranm/repos/spdlog                                                                     
Build dir: /usr/home/vedranm/repos/spdlog/builddir-meson                   
Build type: native build                                                                                           
Program scripts/extract_version.py found: YES (/usr/home/vedranm/repos/spdlog/scripts/extract_version.py)                                
Project name: spdlog                                                                                               
Project version: 1.4.3        
C++ compiler for the host machine: ccache c++ (clang 8.0.1 "FreeBSD clang version 8.0.1 (tags/RELEASE_801/final 366581) (based on LLVM 8.0.1)")                                                                                                                                           
Build machine cpu family: x86_64              
Build machine cpu: x86_64                                                                                          
Run-time dependency threads found: YES 
Found pkg-config: /usr/local/bin/pkg-config (1.6.3)                                                                
Found CMake: /usr/local/bin/cmake (3.15.3)               
Run-time dependency libsystemd found: NO (tried pkgconfig and cmake)                                                                         
Run-time dependency benchmark found: YES 0.0.0                                                                     
Message: spdlog build summary:                                                                                     
                                                                                                                   
  - using external fmt:  false                                                                                     
  - building tests:      true                                                                                      
  - building examples:   true                            
  - building benchmarks: true                                                                                      
  - library type:        static                                                                                    
  - no exceptions:       false                                                                                     
                                                                                                                   
Build targets in project: 9                              
Found ninja-1.9.0 at /usr/local/bin/ninja

% ninja test
[0/1] Running all tests.
1/2 test_spdlog-utests                      FAIL     4.17 s (exit status 2)
2/2 test_spdlog-utests-ho                   FAIL     4.36 s (exit status 1)

Ok:                    0
Expected Fail:         0
Fail:                  2
Unexpected Pass:       0
Skipped:               0
Timeout:               0


The output from the failed tests:

1/2 test_spdlog-utests                      FAIL     4.17 s (exit status 2)

--- command ---
/usr/home/vedranm/repos/spdlog/builddir-meson/tests/spdlog-utests
--- stdout ---

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
spdlog-utests is a Catch v2.8.0 host application.
Run with -? for options

-------------------------------------------------------------------------------
daily_logger rotate
-------------------------------------------------------------------------------
../tests/test_daily_logger.cpp:136
...............................................................................

../tests/test_daily_logger.cpp:133: FAILED:
  REQUIRE( count_files("test_logs") == static_cast<size_t>(expected_n_files) )
with expansion:
  2 == 1

-------------------------------------------------------------------------------
to_file multi-workers
-------------------------------------------------------------------------------
../tests/test_async.cpp:177
...............................................................................

../tests/test_async.cpp:194: FAILED:
  REQUIRE( count_lines(filename) == messages )
with expansion:
  0 == 10240 (0x2800)

[2019-11-03 21:01:49.738] [test] [trace] Test stdout_st
[2019-11-03 21:01:49.738] [test] [debug] Test stdout_mt
[2019-11-03 21:01:49.740] [test] [info] Test stdout_color_st
[2019-11-03 21:01:49.740] [test] [trace] Test stdout_color_mt
===============================================================================
test cases: 112 | 110 passed | 2 failed
assertions: 378 | 376 passed | 2 failed

--- stderr ---
[2019-11-03 21:01:49.739] [test] [info] Test stderr_st
[2019-11-03 21:01:49.739] [test] [info] Test stderr_mt
[2019-11-03 21:01:49.739] [test] [warning] Test stderr_mt
[2019-11-03 21:01:49.739] [test] [error] Test stderr_mt
[2019-11-03 21:01:49.739] [test] [critical] Test stderr_mt
[2019-11-03 21:01:49.740] [test] [debug] Test stderr_color_st
[2019-11-03 21:01:49.740] [test] [info] Test stderr_color_mt
[2019-11-03 21:01:49.740] [test] [warning] Test stderr_color_mt
[2019-11-03 21:01:49.741] [test] [error] Test stderr_color_mt
[2019-11-03 21:01:49.741] [test] [critical] Test stderr_color_mt
[*** LOG ERROR #0001 ***] [2019-11-03 21:01:49] [test-error] {argument index out of range}
-------

2/2 test_spdlog-utests-ho                   FAIL     4.36 s (exit status 1)

--- command ---
/usr/home/vedranm/repos/spdlog/builddir-meson/tests/spdlog-utests-ho
--- stdout ---

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
spdlog-utests-ho is a Catch v2.8.0 host application.
Run with -? for options

-------------------------------------------------------------------------------
to_file multi-workers
-------------------------------------------------------------------------------
../tests/test_async.cpp:177
...............................................................................

../tests/test_async.cpp:194: FAILED:
  REQUIRE( count_lines(filename) == messages )
due to unexpected exception with message:
  Failed open file 

[2019-11-03 21:01:49.991] [test] [trace] Test stdout_st
[2019-11-03 21:01:49.991] [test] [debug] Test stdout_mt
[2019-11-03 21:01:49.991] [test] [info] Test stdout_color_st
[2019-11-03 21:01:49.991] [test] [trace] Test stdout_color_mt
===============================================================================
test cases: 112 | 111 passed | 1 failed
assertions: 383 | 382 passed | 1 failed

--- stderr ---
[2019-11-03 21:01:49.991] [test] [info] Test stderr_st
[2019-11-03 21:01:49.991] [test] [info] Test stderr_mt
[2019-11-03 21:01:49.991] [test] [warning] Test stderr_mt
[2019-11-03 21:01:49.991] [test] [error] Test stderr_mt
[2019-11-03 21:01:49.991] [test] [critical] Test stderr_mt
[2019-11-03 21:01:49.991] [test] [debug] Test stderr_color_st
[2019-11-03 21:01:49.991] [test] [info] Test stderr_color_mt
[2019-11-03 21:01:49.991] [test] [warning] Test stderr_color_mt
[2019-11-03 21:01:49.991] [test] [error] Test stderr_color_mt
[2019-11-03 21:01:49.991] [test] [critical] Test stderr_color_mt
[*** LOG ERROR #0001 ***] [2019-11-03 21:01:50] [test-error] {argument index out of range}
-------

Full log written to /usr/home/vedranm/repos/spdlog/builddir-meson/meson-logs/testlog.txt
FAILED: meson-test 
/usr/local/bin/meson test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
```

**Comment by gabime** [comment]
Does the exact same installation pass under CMake?

**Comment by vedranmiletic** [comment]
Yes, both on Linux and FreeBSD.

**Comment by gabime** [comment]
Fixed in bb1b24c 

Thanks for reporting.

**Comment by vedranmiletic** [comment]
Verified. Thank you.
